client/src/js/containers/syllabus/AddedCourseListContainer.js

```diff
@@ -64,3 +64,9 @@ const mapDispatchToProps = {
 export default connect(mapStateToProps, mapDispatchToProps)(
   AddedCourseListContainer
 );
+
+AddedCourseListContainer.propTypes = {
+  addedCourses: PropTypes.array.isRequired,
+  semesterKey: PropTypes.string.isRequired,
+  changeCoursesSortingOption: PropTypes.func.isRequired
+};
```

Comment:
I didn't mention it in the example, but `mapStateToProps` should also be examined. Basically, `mapDispatchToProps` maps a **dispatch function** to **props**, while `mapStateToProps` maps a slice of the **redux state tree** to **props**. So both of them provide props for the component. If you're not familiar with redux yet, I highly recommend you to learn it with its creator Dan Abramov [here](https://egghead.io/courses/getting-started-with-redux) (it's free).
```javascript
const mapStateToProps = (state, ownProps) => {
  return {
    selectedSortingOption: getSortingOption(
      state.addedCourses[ownProps.semesterKey]
    )
  };
};
```
You should add `selectedSortingOption` in `propTypes` as well

Reply:
`getSortingOption` is **not** a prop. It is a function imported from a reducer and **used to generate the value for `selectedSortingOption`**. `selectedSortingOption` is the prop. Delete `getSortingOption` in `propTypes`.
```javascript
//AddedCourseListContainer.js...
import { getSortingOption } from '../../reducers/addedSemesterCourses';
```

Reply:
The type of `selectedSortingOption` is **not** `object`. Check out `reducers/addedSemesterCourses.js` and try to figure out what data type `getSortingOption` returns.